nixpkgs-unstable: xorg-server fails to build on macos

## Describe the bug

In my nix-darwin setup I am installing `xorg.libX11` which seems to need to build `xorg-server-21.1.14` and it has been failing with the following error for about 2 months:
```
error: builder for '/nix/store/aqhs94v5cq7h5y5186i0igxzm4hjv686-xorg-server-21.1.14.drv' failed with exit code 2;
       last 25 log lines:
       >                         unw_strerror(ret), ret);
       >                         ^
       > backtrace.c:87:25: warning: format specifies type 'char *' but the argument has type 'int' [-Wformat]
       >                         unw_strerror(ret), ret);
       >                         ^~~~~~~~~~~~~~~~~
       > backtrace.c:106:57: error: call to undeclared function 'unw_strerror'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
       >             ErrorFSigSafe("unw_step failed: %s [%d]\n", unw_strerror(ret), ret);
       >                                                         ^
       > backtrace.c:106:57: warning: format specifies type 'char *' but the argument has type 'int' [-Wformat]
       >             ErrorFSigSafe("unw_step failed: %s [%d]\n", unw_strerror(ret), ret);
       >                                             ~~          ^~~~~~~~~~~~~~~~~
       >                                             %d
       > 5 warnings and 6 errors generated.
       > make[1]: *** [Makefile:699: backtrace.lo] Error 1
       > make[1]: *** Waiting for unfinished jobs....
       > io.c:565:33: warning: mixing declarations and code is incompatible with standards before C99 [-Wdeclaration-after-statement]
       >     register ConnectionInputPtr oci = oc->input;
       >                                 ^
       > client.c:171:13: warning: mixing declarations and code is incompatible with standards before C99 [-Wdeclaration-after-statement]
       >         int mib[3];
       >             ^
       > 1 warning generated.
       > 1 warning generated.
       > make[1]: Leaving directory '/private/tmp/nix-build-xorg-server-21.1.14.drv-0/xorg-server-21.1.14/os'
       > make: *** [Makefile:792: all-recursive] Error 1
```

## Steps To Reproduce

Steps to reproduce the behavior:
1. Install `xorg.libX11` from `nixpkgs-unstable` using `nix-darwin`.

## Expected behavior
The package installs properly.


## Additional context
NIx darwin version: `24.11.20241027.0fcb98a+darwin5.b379bd4`

## Metadata
```
- system: `"aarch64-darwin"`
 - host os: `Darwin 24.1.0, macOS 15.1.1`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.24.9`
 - channels(root): `"nixpkgs"`
 - nixpkgs: `/nix/store/93haqkb7bl8l90vpfqgkygrd5l13gafm-source`
```


Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/servers/x11/xorg/overrides.nix
+++ b/pkgs/servers/x11/xorg/overrides.nix
@@ -890,6 +890,7 @@ self: super:
         configureFlags = [
           # note: --enable-xquartz is auto
           "CPPFLAGS=-I${./darwin/dri}"
+          "--disable-libunwind" # libunwind on darwin is missing unw_strerror
           "--disable-glamor"
           "--with-default-font-path="
           "--with-apple-application-name=XQuartz"

